eclipse/kura
# First addition of Maven Archetypes to Eclipse Kura.

Signed-off-by: David Woodard <david.woodard@eurotech.com>

This addresses issue #82.

This is the first attempt at adding an Archetype to Eclipse Kura. This archetype will create a simple example project that uses the CloudService. The project is setup to perform a Maven build to produce a Tycho generated OSGi bundle and an OSGi deployment package.

To test:
Perform a standard Kura build:
mvn -f target-platform/pom.xml clean install
mvn -f kura/manifest_pom.xml clean install -Dmaven.test.skip=true
mvn -f kura/pom_pom.xml clean install -Dmaven.test.skip=true -Pweb,dev-env

This will install the archetype into your local repository. To use the archetype:

mvn archetype:generate -DarchetypeCatalog=local

The project will be created after answering the questions. Make sure to change the the default version value from 1.0-SNAPSHOT to something OSGi friendly (ex: 1.0.0-SNAPSHOT).

We can add other archetypes (or augment this one) to accept a class name for a property. Currently the class name is set to "KuraExample".
